notion_client/client.py

```diff
@@ -102,11 +115,19 @@ def _build_request(
         path: str,
         query: Optional[Dict[Any, Any]] = None,
         body: Optional[Dict[Any, Any]] = None,
-        auth: Optional[str] = None,
+        auth: Optional[Union[str, tuple[str, str]]] = None,
```

Comment:
Feels like a TypedDict would be more appropriate than a tuple here (and closer to the JS SDK implementation), what do you think?